test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,577 @@
+           param ([bool[]] $Parameter ) $Parameter
+        }
+
+        function get-foo22a
+        {
+            param(
+                [System.Reflection.MemberTypes] $memberTypes = $([Enum]::GetNames("System.Reflection.MemberTypes") -join ",")
+            )    
+            $memberTypes
+        }
+    
+        function get-foo22b
+        {
+            [CmdletBinding()]
+            param(
+                [System.Reflection.MemberTypes] $memberTypes = $([Enum]::GetNames("System.Reflection.MemberTypes") -join ",")
+            )    
+            $memberTypes
+        }    
+
+        function get-foo22c
+        {
+            param(
+                [Parameter()]
+                [System.Reflection.MemberTypes] $memberTypes
+            )    
+            $memberTypes
+        }
+
+        function get-foo22d
+        {
+            param(
+                [System.Reflection.MemberTypes] $memberTypes
+            )
+            $memberTypes
+        }
+
+        function get-foo22e
+        {
+            param([ValidateRange(1,42)] $p = 55)
+            $p
+        }
+
+        function get-foo22f
+        {
+            [CmdletBinding()]
+            param([ValidateRange(1,42)] $p = 55)
+            $p
+        }
+
+        function get-foo22g
+        {
+            param([ValidateRange(1,42)] $p)            
+            $p
+        }
+
+        function get-foo22h
+        {
+            [CmdletBinding()]
+            param([ValidateRange(1,42)] $p)
+            $p
+        }
+
+        if(-not $IsCoreCLR)
+        {
+            try {
+                 $null = [Test.Language.ParameterBinding.MyClass]
+            }
+            catch {
+                add-type -PassThru -TypeDefinition @'
+                using System.Management.Automation;
+                using System;
+                using System.Collections;
+                using System.Collections.ObjectModel;
+                using System.IO;
+
+                namespace Test.Language.ParameterBinding {
+                    public class MyClass : Collection<string>
+                    {
+                        public MyClass() {}
+                        public MyClass(Hashtable h) {}
+                    }
+
+                    [Cmdlet("Get", "TestCmdlet")]
+                    public class MyCmdlet : PSCmdlet {
+                        [Parameter]
+                        public MyClass MyParameter
+                        {
+                            get { return myParameter; }
+                            set { myParameter = value; }
+                        }
+                        private MyClass myParameter;
+
+                        protected override void ProcessRecord()
+                        {
+                            WriteObject((myParameter == null) ? "<null>" : "hashtable");
+                        }
+                    }
+                }
+'@ | % {$_.assembly} | Import-module
+            }
+        }
+    }
+
+    $testData = @(
```

Comment:
I made the below changes. Please review:
1. For those functions being reused by different variations, I call Validate-Context where the function is passed as Setup parameter and is called in context and the validation is done in the It block which takes the variation as test cases.
2. For those functions being used by only one tests, we pass the function as Setup item and is called in the It block.


---
In reply to: [78436715](https://github.com/Powershell/Powershell/pull/2218#discussion_r78436715) [](ancestors = 78436715)